test-common/utils.c

```diff
@@ -2042,10 +2084,37 @@ flatpak_install_info_new (FlatpakInstallInfoType  type,
   info->runtime_branch = g_strdup (runtime_branch);
   info->repo_name = g_strdup (repo_name);
   info->preinstall = preinstall;
+  info->extension_of_ref = g_strdup (extension_of_ref);
+  info->extension_infos = extension_infos != NULL ?
+    g_ptr_array_ref (extension_infos) :
+    g_ptr_array_new_with_free_func ((GDestroyNotify) flatpak_extension_point_info_free);
 
   return info;
 }
 
+FlatpakInstallInfo *
+flatpak_install_info_new (FlatpakInstallInfoType  type,
+                          const gchar            *name,
+                          const gchar            *branch,
+                          const gchar            *runtime_name,
+                          const gchar            *runtime_branch,
+                          const gchar            *repo_name,
+                          gboolean                preinstall)
+{
+  g_autoptr(GPtrArray) extension_infos =
+    g_ptr_array_new_with_free_func ((GDestroyNotify) flatpak_extension_point_info_free);
```

Comment:
Instead of passing an empty `GPtrArray` around, you could just accept `NULL` as equivalent to an empty array in the extension info functions above.